src/zxing/zxing/qrcode/encoder/MaskUtil.cpp

```diff
@@ -169,7 +169,7 @@ bool MaskUtil::getDataMaskBit(int maskPattern, int x, int y)
         intermediate = ((temp % 3) + ((y + x) & 0x1)) & 0x1;
         break;
     default:
-        throw IllegalArgumentException("Invalid mask pattern: " + maskPattern);
+        throw IllegalArgumentException("Invalid mask pattern");
```

Comment:
This won't work: the string returned by stringstream will be destroyed when leaving the ZXingStringWithNumber() function, so when printing exception information, a deleted string will be referenced.

I suggest changing the **message** field of Exception to std::string, and then formatting messages using stringstream at the site which throws the exception, so it would become:

`
std::stringstream strm;
strm << "Invalid mask pattern: " << maskPattern;
throw IllegalArgumentException(strm.str());
`
This way, the caller can do all kinds of formatting.